File: backend/src/seed.ts
// backend/src/seed.ts
import mongoose from 'mongoose';
import dotenv from 'dotenv';
import { User } from './models/User';
import { Board } from './models/Board';
import { Column } from './models/Column';
import { Task } from './models/Task';
import { AuthUtils } from './utils/auth';
import { Role, UserStatus, BoardType, Template, Priority, Department } from './types';
import { RoleModel } from './models/Role';

dotenv.config();

// ─────────────────────────────────────────────────────────────────────────────
// Helpers
// ─────────────────────────────────────────────────────────────────────────────
const ensureRole = async (key: string, name?: string, builtIn = true) => {
  const KEY = key.trim().toUpperCase();
  await RoleModel.updateOne(
    { key: KEY },
    { $setOnInsert: { name: name ?? KEY, builtIn, isActive: true } },
    { upsert: true }
  );
};

// ─────────────────────────────────────────────────────────────────────────────
// Seed
// ─────────────────────────────────────────────────────────────────────────────
const seedData = async (): Promise<void> => {
  try {
    // Connect
    const mongoURL = process.env.MONGO_URL;
    if (!mongoURL) throw new Error('MONGO_URL environment variable is not defined');
    const dbName = process.env.DB_NAME || 'simplified_jira';

    await mongoose.connect(mongoURL, { dbName });
    console.log(`✅ Connected to MongoDB database: ${dbName}`);

    // Optional: wipe roles too (controlled by env)
    if (String(process.env.SEED_RESET_ROLES).toLowerCase() === 'true') {
      console.log('🧹 Clearing roles (by SEED_RESET_ROLES=true)...');
      await RoleModel.deleteMany({});
      console.log('✓ Cleared roles');
    }

    // Always wipe core data
    console.log('🧹 Clearing core collections...');
    await User.deleteMany({});
    await Board.deleteMany({});
    await Column.deleteMany({});
    await Task.deleteMany({});
    console.log('✓ Cleared users, boards, columns, tasks');

    // Base roles
    console.log('🔑 Ensuring base roles...');
    await ensureRole('ADMIN', 'Admin');
    await ensureRole('TECH', 'Tech');
    await ensureRole('DESIGNER', 'Designer');
    await ensureRole('BUYER', 'Buyer');
    await ensureRole('TEAM_LEAD', 'Team Lead'); // базовая роль тимлида
    console.log('✓ Base roles ready');

    // Users
    console.log('👥 Creating users...');
    const users = [
      {
        id: 'admin-001',
        email: 'admin@company.com',
        passwordHash: await AuthUtils.hashPassword('admin123'),
        fullName: 'Admin User',
        roles: [Role.ADMIN], // 'admin'
        status: UserStatus.ACTIVE,
      },
      {
        id: 'lead-001',
        email: 'lead@company.com',
        passwordHash: await AuthUtils.hashPassword('lead123'),
        fullName: 'Team Lead',
        roles: [Role.TEAM_LEAD], // 'team_lead'
        status: UserStatus.ACTIVE,
      },
      {
        id: 'buyer-001',
        email: 'buyer@company.com',
        passwordHash: await AuthUtils.hashPassword('buyer123'),
        fullName: 'Alice Buyer',
        roles: [Role.BUYER],
        department: Department.SWIP,
        status: UserStatus.ACTIVE,
      },
      {
        id: 'buyer-002',
        email: 'buyer2@company.com',
        passwordHash: await AuthUtils.hashPassword('buyer123'),
        fullName: 'Bob Buyer',
        roles: [Role.BUYER],
        department: Department.GAMBLING,
        status: UserStatus.ACTIVE,
      },
      {
        id: 'designer-001',
        email: 'designer@company.com',
        passwordHash: await AuthUtils.hashPassword('designer123'),
        fullName: 'Charlie Designer',
        roles: [Role.DESIGNER],
        department: Department.GAMBLING,
        status: UserStatus.ACTIVE,
      },
      {
        id: 'tech-001',
        email: 'tech@company.com',
        passwordHash: await AuthUtils.hashPassword('tech123'),
        fullName: 'David Tech',
        roles: [Role.TECH],
        status: UserStatus.ACTIVE,
      },
    ];
    await User.insertMany(users);
    console.log(`✓ Created ${users.length} users`);

    // Boards (добавил team_lead в allowedRoles, чтобы тимлид видел все борды)
    console.log('📋 Creating boards...');
    const boards = [
      {
        id: 'board-buyers',
        name: 'Buyers',
        key: 'BUY',
        type: BoardType.TASKS,
        template: Template.KANBAN_BASIC,
        allowedRoles: [Role.BUYER, Role.ADMIN, Role.TEAM_LEAD],
        owners: ['buyer-001'],
      },
      {
        id: 'board-designers',
        name: 'Designers',
        key: 'DES',
        type: BoardType.TASKS,
        template: Template.KANBAN_BASIC,
        allowedRoles: [Role.DESIGNER, Role.ADMIN, Role.TEAM_LEAD],
        owners: ['designer-001'],
        // department: Department.GAMBLING,
      },
      {
        id: 'board-tech',
        name: 'Tech',
        key: 'TECH',
        type: BoardType.TASKS,
        template: Template.KANBAN_TJ_TECH,
        allowedRoles: [Role.TECH, Role.ADMIN, Role.TEAM_LEAD],
        owners: ['tech-001'],
      },
      {
        id: 'board-expenses',
        name: 'Expenses',
        key: 'EXP',
        type: BoardType.EXPENSES,
        template: Template.KANBAN_BASIC,
        allowedRoles: [Role.ADMIN, Role.BUYER, Role.TEAM_LEAD],
        owners: ['admin-001'],
      },
    ];
    await Board.insertMany(boards);
    console.log(`✓ Created ${boards.length} boards`);

    // Columns
    console.log('📑 Creating columns...');
    const columns = [
      // Buyers
      { id: 'col-buy-backlog', boardId: 'board-buyers', key: 'BACKLOG', name: 'Backlog', order: 1 },
      { id: 'col-buy-progress', boardId: 'board-buyers', key: 'IN_PROGRESS', name: 'In Progress', order: 2 },
      { id: 'col-buy-to-tech', boardId: 'board-buyers', key: 'TO_TECH', name: 'To Tech', order: 3 },
      { id: 'col-buy-to-designers', boardId: 'board-buyers', key: 'TO_DESIGNERS', name: 'To Designers', order: 4 },
      { id: 'col-buy-done', boardId: 'board-buyers', key: 'DONE', name: 'Done', order: 5 },

      // Designers
      { id: 'col-des-queue', boardId: 'board-designers', key: 'QUEUE', name: 'Queue', order: 1 },
      { id: 'col-des-doing', boardId: 'board-designers', key: 'DOING', name: 'Doing', order: 2 },
      { id: 'col-des-review', boardId: 'board-designers', key: 'REVIEW', name: 'Review', order: 3 },
      { id: 'col-des-done', boardId: 'board-designers', key: 'DONE', name: 'Done', order: 4 },

      // Tech
      { id: 'col-tech-todo', boardId: 'board-tech', key: 'TODO', name: 'Todo', order: 1 },
      { id: 'col-tech-dev', boardId: 'board-tech', key: 'IN_DEV', name: 'In Dev', order: 2 },
      { id: 'col-tech-review', boardId: 'board-tech', key: 'CODE_REVIEW', name: 'Code Review', order: 3 },
      { id: 'col-tech-done', boardId: 'board-tech', key: 'DONE', name: 'Done', order: 4 },

      // Expenses
      { id: 'col-exp-pending', boardId: 'board-expenses', key: 'PENDING', name: 'Pending', order: 1 },
      { id: 'col-exp-approved', boardId: 'board-expenses', key: 'APPROVED', name: 'Approved', order: 2 },
      { id: 'col-exp-paid', boardId: 'board-expenses', key: 'PAID', name: 'Paid', order: 3 },
    ];
    await Column.insertMany(columns);
    console.log(`✓ Created ${columns.length} columns`);

    // Helpers
    const u = (id: string) => users.find(x => x.id === id)!;
    const deptOf = (id: string) => (u(id) as any).department;

    // Tasks
    console.log('📝 Creating tasks...');
    const tasks = [
      // Alice (buyer-001, SWIP)
      {
        id: 'task-buy-001',
        boardKey: 'BUY',
        columnId: 'col-buy-backlog',
        title: 'Research new vendor options (Alice)',
        description: 'Alice needs to find alternative suppliers for Q2',
        priority: Priority.HIGH,
        tags: ['research', 'vendors', 'alice'],
        dueDate: new Date('2025-02-15'),
        assigneeId: 'buyer-001',
        creatorId: 'buyer-001',
        department: deptOf('buyer-001'),
      },
      {
        id: 'task-buy-002',
        boardKey: 'BUY',
        columnId: 'col-buy-progress',
        title: 'Budget approval for new tools (Alice)',
        description: 'Alice is getting approval for design software licenses',
        priority: Priority.MEDIUM,
        tags: ['budget', 'tools', 'alice'],
        assigneeId: 'buyer-001',
        creatorId: 'buyer-001',
        department: deptOf('buyer-001'),
      },

      // Bob (buyer-002, GAMBLING)
      {
        id: 'task-buy-005',
        boardKey: 'BUY',
        columnId: 'col-buy-backlog',
        title: 'Office supplies procurement (Bob)',
        description: 'Bob needs to order office supplies for next quarter',
        priority: Priority.MEDIUM,
        tags: ['office', 'supplies', 'bob'],
        dueDate: new Date('2025-02-20'),
        assigneeId: 'buyer-002',
        creatorId: 'buyer-002',
        department: deptOf('buyer-002'),
      },
      {
        id: 'task-buy-006',
        boardKey: 'BUY',
        columnId: 'col-buy-progress',
        title: 'Equipment lease negotiations (Bob)',
        description: 'Bob is negotiating lease terms for new equipment',
        priority: Priority.HIGH,
        tags: ['equipment', 'lease', 'bob'],
        dueDate: new Date('2025-02-18'),
        assigneeId: 'buyer-002',
        creatorId: 'buyer-002',
        department: deptOf('buyer-002'),
      },

      // Cross-team routing
      {
        id: 'task-buy-003',
        boardKey: 'BUY',
        columnId: 'col-buy-to-tech',
        title: 'Technical requirements for API integration',
        description: 'Need tech team to define API specs for new vendor system',
        priority: Priority.HIGH,
        tags: ['api', 'integration', 'cross-team'],
        dueDate: new Date('2025-02-10'),
        assigneeId: 'tech-001',
        creatorId: 'buyer-001',
        department: deptOf('buyer-001'),
      },
      {
        id: 'task-buy-004',
        boardKey: 'BUY',
        columnId: 'col-buy-to-designers',
        title: 'UI mockups for vendor dashboard',
        description: 'Need designs for the new vendor management interface',
        priority: Priority.MEDIUM,
        tags: ['ui', 'dashboard', 'cross-team'],
        dueDate: new Date('2025-02-20'),
        assigneeId: 'designer-001',
        creatorId: 'buyer-001',
        department: deptOf('buyer-001'),
      },

      // Designer (designer-001, GAMBLING)
      {
        id: 'task-des-001',
        boardKey: 'DES',
        columnId: 'col-des-queue',
        title: 'Landing page redesign',
        description: 'Redesign the main landing page with new branding',
        priority: Priority.HIGH,
        tags: ['landing', 'branding'],
        dueDate: new Date('2025-02-25'),
        assigneeId: 'designer-001',
        creatorId: 'designer-001',
        department: deptOf('designer-001'),
      },
      {
        id: 'task-des-002',
        boardKey: 'DES',
        columnId: 'col-des-doing',
        title: 'Mobile app icon set',
        description: 'Create consistent icon set for mobile application',
        priority: Priority.MEDIUM,
        tags: ['mobile', 'icons'],
        assigneeId: 'designer-001',
        creatorId: 'designer-001',
        department: deptOf('designer-001'),
      },

      // Tech (tech-001)
      {
        id: 'task-tech-001',
        boardKey: 'TECH',
        columnId: 'col-tech-todo',
        title: 'Database migration script',
        description: 'Create migration for new user permissions table',
        priority: Priority.HIGH,
        tags: ['database', 'migration'],
        dueDate: new Date('2025-02-12'),
        assigneeId: 'tech-001',
        creatorId: 'tech-001',
      },
      {
        id: 'task-tech-002',
        boardKey: 'TECH',
        columnId: 'col-tech-dev',
        title: 'Authentication service refactor',
        description: 'Improve JWT token handling and add refresh tokens',
        priority: Priority.MEDIUM,
        tags: ['auth', 'refactor'],
        assigneeId: 'tech-001',
        creatorId: 'tech-001',
      },
      {
        id: 'task-tech-003',
        boardKey: 'TECH',
        columnId: 'col-tech-review',
        title: 'API rate limiting implementation',
        description: 'Add rate limiting middleware to prevent abuse',
        priority: Priority.LOW,
        tags: ['api', 'security'],
        dueDate: new Date('2025-03-01'),
        assigneeId: 'tech-001',
        creatorId: 'tech-001',
      },

      // Expenses
      {
        id: 'task-exp-001',
        boardKey: 'EXP',
        columnId: 'col-exp-pending',
        title: 'Office Supplies - Q1 2025',
        description: 'Purchase necessary office supplies for the first quarter',
        priority: Priority.MEDIUM,
        tags: ['office', 'supplies'],
        amount: 1250.5,
        assigneeId: 'admin-001',
        creatorId: 'buyer-001',
      },
      {
        id: 'task-exp-002',
        boardKey: 'EXP',
        columnId: 'col-exp-approved',
        title: 'Software Licenses - Figma Team',
        description: 'Annual Figma team licenses for design team',
        priority: Priority.HIGH,
        tags: ['software', 'licenses'],
        amount: 4800.0,
        assigneeId: 'admin-001',
        creatorId: 'designer-001',
      },
      {
        id: 'task-exp-003',
        boardKey: 'EXP',
        columnId: 'col-exp-paid',
        title: 'Server Infrastructure - AWS',
        description: 'Monthly AWS infrastructure costs',
        priority: Priority.HIGH,
        tags: ['infrastructure', 'aws'],
        amount: 2340.75,
        assigneeId: 'admin-001',
        creatorId: 'tech-001',
      },
      {
        id: 'task-exp-004',
        boardKey: 'EXP',
        columnId: 'col-exp-pending',
        title: 'Team Building Event',
        description: 'Annual team building retreat expenses',
        priority: Priority.LOW,
        tags: ['team', 'event'],
        amount: 8500.0,
        assigneeId: 'buyer-001',
        creatorId: 'admin-001',
      },
    ];

    await Task.insertMany(tasks);
    console.log(`✓ Created ${tasks.length} tasks`);

    // Verify
    const [uCount, bCount, cCount, tCount] = await Promise.all([
      User.countDocuments({}),
      Board.countDocuments({}),
      Column.countDocuments({}),
      Task.countDocuments({}),
    ]);
    console.log('\n✓ Final verification:');
    console.log(`Users: ${uCount}, Boards: ${bCount}, Columns: ${cCount}, Tasks: ${tCount}`);

    console.log('\n🎉 Seed completed successfully!');
    console.log('\nLogin credentials:');
    console.log('Admin: admin@company.com / admin123');
    console.log('Team Lead: lead@company.com / lead123');
    console.log('Alice (Buyer SWIP): buyer@company.com / buyer123');
    console.log('Bob (Buyer GAMBLING): buyer2@company.com / buyer123');
    console.log('Charlie (Designer): designer@company.com / designer123');
    console.log('David (Tech): tech@company.com / tech123');

    await mongoose.connection.close();
    process.exit(0);
  } catch (error) {
    console.error('❌ Seed failed:', error);
    try { await mongoose.connection.close(); } catch {}
    process.exit(1);
  }
};

seedData();

File: backend/seed.py
#!/usr/bin/env python3
"""
Seed script for Simplified Jira application
Creates initial users, boards, columns, and tasks
"""
import asyncio
import os
from motor.motor_asyncio import AsyncIOMotorClient
from passlib.hash import bcrypt
from datetime import datetime, timezone
from dotenv import load_dotenv
from pathlib import Path

ROOT_DIR = Path(__file__).parent
load_dotenv(ROOT_DIR / '.env')

# MongoDB connection
mongo_url = os.environ['MONGO_URL']
client = AsyncIOMotorClient(mongo_url)
db = client[os.environ['DB_NAME']]

async def clear_collections():
    """Clear existing data"""
    print("Clearing existing data...")
    await db.users.delete_many({})
    await db.boards.delete_many({})
    await db.columns.delete_many({})
    await db.tasks.delete_many({})
    print("✓ Cleared all collections")

async def create_users():
    """Create seed users"""
    print("Creating users...")
    
    users = [
        {
            "id": "admin-001",
            "email": "admin@company.com",
            "password_hash": bcrypt.hash("admin123"),
            "full_name": "Admin User",
            "roles": ["admin"],
            "groups": [],
            "created_at": datetime.now(timezone.utc),
            "updated_at": datetime.now(timezone.utc)
        },
        {
            "id": "buyer-001",
            "email": "buyer@company.com",
            "password_hash": bcrypt.hash("buyer123"),
            "full_name": "Alice Buyer",
            "roles": ["buyer"],
            "groups": [],
            "created_at": datetime.now(timezone.utc),
            "updated_at": datetime.now(timezone.utc)
        },
        {
            "id": "buyer-002",
            "email": "buyer2@company.com",
            "password_hash": bcrypt.hash("buyer123"),
            "full_name": "Bob Buyer",
            "roles": ["buyer"],
            "groups": [],
            "created_at": datetime.now(timezone.utc),
            "updated_at": datetime.now(timezone.utc)
        },
        {
            "id": "designer-001",
            "email": "designer@company.com",
            "password_hash": bcrypt.hash("designer123"),
            "full_name": "Charlie Designer", 
            "roles": ["designer"],
            "groups": [],
            "created_at": datetime.now(timezone.utc),
            "updated_at": datetime.now(timezone.utc)
        },
        {
            "id": "tech-001",
            "email": "tech@company.com",
            "password_hash": bcrypt.hash("tech123"),
            "full_name": "David Tech",
            "roles": ["tech"],
            "groups": [],
            "created_at": datetime.now(timezone.utc),
            "updated_at": datetime.now(timezone.utc)
        },
        {
            "id": "vladislav-001",
            "email": "vladislav@company.com",
            "password_hash": bcrypt.hash("vladislav123"),
            "full_name": "Vladislav",
            "roles": ["main_admin"],  # Main admin for expenses
            "groups": [],
            "created_at": datetime.now(timezone.utc),
            "updated_at": datetime.now(timezone.utc)
        },
        {
            "id": "olya-coo",
            "email": "olya@company.com",
            "password_hash": bcrypt.hash("olya123"),
            "full_name": "Olya COO",
            "roles": ["main_admin"],  # Main admin for expenses
            "groups": [],
            "created_at": datetime.now(timezone.utc),
            "updated_at": datetime.now(timezone.utc)
        },
        {
            "id": "lead-001",
            "email": "lead@company.com",
            "password_hash": bcrypt.hash("lead123"),
            "full_name": "Team Lead",
            "roles": ["lead"],  # Lead sees all buyer tasks
            "groups": [],
            "created_at": datetime.now(timezone.utc),
            "updated_at": datetime.now(timezone.utc)
        }
    ]
    
    await db.users.insert_many(users)
    print(f"✓ Created {len(users)} users")

async def create_boards():
    """Create seed boards"""
    print("Creating boards...")
    
    boards = [
        {
            "id": "board-buyers",
            "name": "Buyers",
            "key": "BUY",
            "type": "tasks",
            "template": "kanban-basic",
            "is_archived": False,
            "settings": {"assignee_enabled": True},
            "allowed_roles": ["buyer", "admin", "lead"],  # Include lead
            "allowed_group_ids": [],
            "members": [],
            "owners": ["buyer-001"],
            "created_at": datetime.now(timezone.utc),
            "updated_at": datetime.now(timezone.utc)
        },
        {
            "id": "board-designers",
            "name": "Designers",
            "key": "DES",
            "type": "tasks", 
            "template": "kanban-basic",
            "is_archived": False,
            "settings": {"assignee_enabled": True},
            "allowed_roles": ["designer", "admin", "lead"],  # Include lead
            "allowed_group_ids": [],
            "members": [],
            "owners": ["designer-001"],
            "created_at": datetime.now(timezone.utc),
            "updated_at": datetime.now(timezone.utc)
        },
        {
            "id": "board-tech",
            "name": "Tech",
            "key": "TECH",
            "type": "tasks",
            "template": "kanban-tj-tech", 
            "is_archived": False,
            "settings": {"assignee_enabled": True},
            "allowed_roles": ["tech", "admin", "lead"],  # Include lead
            "allowed_group_ids": [],
            "members": [],
            "owners": ["tech-001"],
            "created_at": datetime.now(timezone.utc),
            "updated_at": datetime.now(timezone.utc)
        },
        {
            "id": "board-expenses",
            "name": "Expenses",
            "key": "EXPENSES",
            "type": "expenses",
            "template": "kanban-basic", 
            "is_archived": False,
            "settings": {"assignee_enabled": True},
            "allowed_roles": ["admin", "main_admin", "lead", "buyer", "designer", "tech"],  # Include all roles
            "allowed_group_ids": [],
            "members": [],
            "owners": ["admin-001"],  # Only admin can own
            "created_at": datetime.now(timezone.utc),
            "updated_at": datetime.now(timezone.utc)
        }
    ]
    
    await db.boards.insert_many(boards)
    print(f"✓ Created {len(boards)} boards")

async def create_columns():
    """Create seed columns"""
    print("Creating columns...")
    
    columns = [
        # Buyers board columns
        {"id": "col-buy-backlog", "board_id": "board-buyers", "key": "BACKLOG", "name": "Backlog", "order": 1},
        {"id": "col-buy-progress", "board_id": "board-buyers", "key": "IN_PROGRESS", "name": "In Progress", "order": 2},
        {"id": "col-buy-done", "board_id": "board-buyers", "key": "DONE", "name": "Done", "order": 3},
        
        # Designers board columns  
        {"id": "col-des-queue", "board_id": "board-designers", "key": "QUEUE", "name": "Queue", "order": 1},
        {"id": "col-des-doing", "board_id": "board-designers", "key": "DOING", "name": "Doing", "order": 2},
        {"id": "col-des-review", "board_id": "board-designers", "key": "REVIEW", "name": "Review", "order": 3},
        {"id": "col-des-done", "board_id": "board-designers", "key": "DONE", "name": "Done", "order": 4},
        
        # Tech board columns
        {"id": "col-tech-todo", "board_id": "board-tech", "key": "TODO", "name": "Todo", "order": 1},
        {"id": "col-tech-dev", "board_id": "board-tech", "key": "IN_DEV", "name": "In Dev", "order": 2},
        {"id": "col-tech-review", "board_id": "board-tech", "key": "CODE_REVIEW", "name": "Code Review", "order": 3},
        {"id": "col-tech-done", "board_id": "board-tech", "key": "DONE", "name": "Done", "order": 4},
        
        # Expenses board columns
        {"id": "col-exp-waiting", "board_id": "board-expenses", "key": "WAITING", "name": "Ожидание", "order": 1},
        {"id": "col-exp-progress", "board_id": "board-expenses", "key": "IN_PROGRESS", "name": "In Progress", "order": 2},
        {"id": "col-exp-paid", "board_id": "board-expenses", "key": "PAID", "name": "Paid", "order": 3},
    ]
    
    # Add timestamps
    for col in columns:
        col["created_at"] = datetime.now(timezone.utc)
        col["updated_at"] = datetime.now(timezone.utc)
    
    await db.columns.insert_many(columns)
    print(f"✓ Created {len(columns)} columns")

async def create_tasks():
    """Create seed tasks"""
    print("Creating tasks...")
    
    tasks = [
        # Buyers board tasks - Alice's tasks
        {
            "id": "task-buy-001",
            "board_key": "BUY",
            "column_id": "col-buy-backlog",
            "title": "Research new vendor options (Alice)",
            "description": "Alice needs to find alternative suppliers for Q2",
            "priority": "high",
            "tags": ["research", "vendors", "alice"],
            "due_date": "2025-02-15",
            "assignee_id": "buyer-001",
            "creator_id": "buyer-001"
        },
        {
            "id": "task-buy-002", 
            "board_key": "BUY",
            "column_id": "col-buy-progress",
            "title": "Budget approval for new tools (Alice)",
            "description": "Alice is getting approval for design software licenses",
            "priority": "medium",
            "tags": ["budget", "tools", "alice"],
            "due_date": None,
            "assignee_id": "buyer-001",
            "creator_id": "buyer-001"
        },
        # Bob's tasks
        {
            "id": "task-buy-005",
            "board_key": "BUY",
            "column_id": "col-buy-backlog",
            "title": "Office supplies procurement (Bob)",
            "description": "Bob needs to order office supplies for next quarter",
            "priority": "medium",
            "tags": ["office", "supplies", "bob"],
            "due_date": "2025-02-20",
            "assignee_id": "buyer-002",
            "creator_id": "buyer-002"
        },
        {
            "id": "task-buy-006", 
            "board_key": "BUY",
            "column_id": "col-buy-progress",
            "title": "Equipment lease negotiations (Bob)",
            "description": "Bob is negotiating lease terms for new equipment",
            "priority": "high",
            "tags": ["equipment", "lease", "bob"],
            "due_date": "2025-02-18",
            "assignee_id": "buyer-002",
            "creator_id": "buyer-002"
        },
        {
            "id": "task-buy-003",
            "board_key": "BUY",
            "column_id": "col-buy-to-tech", 
            "title": "Technical requirements for API integration",
            "description": "Need tech team to define API specs for new vendor system",
            "priority": "high",
            "tags": ["api", "integration", "cross-team"],
            "due_date": "2025-02-10",
            "assignee_id": "tech-001",
            "creator_id": "buyer-001"
        },
        {
            "id": "task-buy-004",
            "board_key": "BUY",
            "column_id": "col-buy-to-designers",
            "title": "UI mockups for vendor dashboard", 
            "description": "Need designs for the new vendor management interface",
            "priority": "medium",
            "tags": ["ui", "dashboard", "cross-team"],
            "due_date": "2025-02-20",
            "assignee_id": "designer-001",
            "creator_id": "buyer-001"
        },
        
        # Designers board tasks
        {
            "id": "task-des-001",
            "board_key": "DES", 
            "column_id": "col-des-queue",
            "title": "Landing page redesign",
            "description": "Redesign the main landing page with new branding",
            "priority": "high",
            "tags": ["landing", "branding"],
            "due_date": "2025-02-25",
            "assignee_id": "designer-001",
            "creator_id": "designer-001"
        },
        {
            "id": "task-des-002",
            "board_key": "DES",
            "column_id": "col-des-doing", 
            "title": "Mobile app icon set",
            "description": "Create consistent icon set for mobile application",
            "priority": "medium",
            "tags": ["mobile", "icons"],
            "due_date": None,
            "assignee_id": "designer-001",
            "creator_id": "designer-001"
        },
        
        # Tech board tasks
        {
            "id": "task-tech-001",
            "board_key": "TECH",
            "column_id": "col-tech-todo",
            "title": "Database migration script",
            "description": "Create migration for new user permissions table", 
            "priority": "high",
            "tags": ["database", "migration"],
            "due_date": "2025-02-12",
            "assignee_id": "tech-001",
            "creator_id": "tech-001"
        },
        {
            "id": "task-tech-002",
            "board_key": "TECH",
            "column_id": "col-tech-dev",
            "title": "Authentication service refactor",
            "description": "Improve JWT token handling and add refresh tokens",
            "priority": "medium", 
            "tags": ["auth", "refactor"],
            "due_date": None,
            "assignee_id": "tech-001",
            "creator_id": "tech-001"
        },
        {
            "id": "task-tech-003",
            "board_key": "TECH",
            "column_id": "col-tech-review",
            "title": "API rate limiting implementation",
            "description": "Add rate limiting middleware to prevent abuse",
            "priority": "low",
            "tags": ["api", "security"],
            "due_date": "2025-03-01",
            "assignee_id": "tech-001", 
            "creator_id": "tech-001"
        },
        
        # Expenses board tasks - Olya and Vladislav
        {
            "id": "task-exp-001",
            "board_key": "EXPENSES",
            "column_id": "col-exp-waiting",
            "title": "Офисная мебель (Olya)",
            "description": "Покупка новых стульев для офиса",
            "priority": "medium",
            "tags": ["офис", "мебель", "olya"],
            "due_date": "2025-03-01",
            "assignee_id": "olya-coo",
            "creator_id": "olya-coo",
            "amount": 15000,
            "category": "office_supplies"
        },
        {
            "id": "task-exp-002",
            "board_key": "EXPENSES",
            "column_id": "col-exp-waiting", 
            "title": "Программное обеспечение (Vladislav)",
            "description": "Лицензия на Adobe Creative Suite",
            "priority": "high",
            "tags": ["ПО", "лицензия", "vladislav"],
            "due_date": "2025-02-28",
            "assignee_id": "vladislav-001",
            "creator_id": "vladislav-001",
            "amount": 25000,
            "category": "software"
        },
        {
            "id": "task-exp-003",
            "board_key": "EXPENSES",
            "column_id": "col-exp-processing",
            "title": "Командировка в Москву (Olya)",
            "description": "Деловая поездка для переговоров с клиентом",
            "priority": "high",
            "tags": ["командировка", "москва", "olya"],
            "due_date": "2025-02-25",
            "assignee_id": "olya-coo",
            "creator_id": "olya-coo", 
            "amount": 45000,
            "category": "travel"
        },
        {
            "id": "task-exp-004",
            "board_key": "EXPENSES",
            "column_id": "col-exp-approved",
            "title": "Канцтовары (Vladislav)",
            "description": "Заказ канцелярских принадлежностей на квартал",
            "priority": "low",
            "tags": ["канцтовары", "квартал", "vladislav"],
            "due_date": None,
            "assignee_id": "vladislav-001",
            "creator_id": "vladislav-001",
            "amount": 8500,
            "category": "office_supplies"
        }
    ]
    
    # Add timestamps
    for task in tasks:
        task["created_at"] = datetime.now(timezone.utc)
        task["updated_at"] = datetime.now(timezone.utc)
    
    await db.tasks.insert_many(tasks)
    print(f"✓ Created {len(tasks)} tasks")

async def main():
    """Main seeding function"""
    print("🌱 Starting seed process...")
    
    try:
        await clear_collections()
        await create_users()
        await create_boards()
        await create_columns()
        await create_tasks()
        
        print("\n✅ Seed completed successfully!")
        print("\nLogin credentials:")
        print("Admin: admin@company.com / admin123")
        print("Alice (Buyer): buyer@company.com / buyer123") 
        print("Bob (Buyer): buyer2@company.com / buyer123")
        print("Team Lead: lead@company.com / lead123")
        print("Olya COO: olya@company.com / olya123") 
        print("Vladislav: vladislav@company.com / vladislav123")
        print("Charlie (Designer): designer@company.com / designer123")
        print("David (Tech): tech@company.com / tech123")
        
    except Exception as e:
        print(f"❌ Seed failed: {e}")
        raise
    finally:
        client.close()

if __name__ == "__main__":
    asyncio.run(main())